esp-hal-common/src/rtc_cntl/rtc/esp32c6.rs

```diff
@@ -315,3 +196,500 @@ pub enum SocResetReason {
+                RtcCalSel::RtcCalRtcMux => match RtcClock::get_slow_freq() {
+                    RtcSlowClock::RtcSlowClock32kXtal => RtcCalSel::RtcCal32kXtal,
+                    RtcSlowClock::RtcSlowClock32kRc => RtcCalSel::RtcCal32kRc,
+                    _ => cal_clk,
+                },
+                RtcCalSel::RtcCal32kOscSlow => RtcCalSel::RtcCalRtcMux,
+                _ => cal_clk,
+            };
+        }
+
+        let lp_clkrst = unsafe { &*LP_CLKRST::ptr() };
+        let pcr = unsafe { &*PCR::ptr() };
+        let pmu = unsafe { &*PMU::ptr() };
+
+        let clk_src = RtcClock::get_slow_freq();
+
+        if cal_clk == RtcCalSel::RtcCalRtcMux {
+            cal_clk = match clk_src {
+                RtcSlowClock::RtcSlowClockRcSlow => RtcCalSel::RtcCalRcSlow,
+                RtcSlowClock::RtcSlowClock32kXtal => RtcCalSel::RtcCal32kXtal,
+                RtcSlowClock::RtcSlowClock32kRc => RtcCalSel::RtcCal32kRc,
+                RtcSlowClock::RtcSlowOscSlow => RtcCalSel::RtcCal32kOscSlow,
+            };
+        }
+
+        let cali_clk_sel;
+        if cal_clk == RtcCalSel::RtcCalRtcMux {
+            cal_clk = match clk_src {
+                RtcSlowClock::RtcSlowClockRcSlow => RtcCalSel::RtcCalRcSlow,
+                RtcSlowClock::RtcSlowClock32kXtal => RtcCalSel::RtcCal32kXtal,
+                RtcSlowClock::RtcSlowClock32kRc => RtcCalSel::RtcCal32kRc,
+                RtcSlowClock::RtcSlowOscSlow => RtcCalSel::RtcCalRcSlow,
+            }
+        }
+
+        if cal_clk == RtcCalSel::RtcCalRcFast {
+            cali_clk_sel = RtcCaliClkSel::CaliClkRcFast;
+        } else if cal_clk == RtcCalSel::RtcCalRcSlow {
+            cali_clk_sel = RtcCaliClkSel::CaliClkRcSlow;
+        } else {
+            cali_clk_sel = RtcCaliClkSel::CaliClk32k;
+            match cal_clk {
+                RtcCalSel::RtcCalRtcMux | RtcCalSel::RtcCalRcSlow | RtcCalSel::RtcCalRcFast => (),
+                RtcCalSel::RtcCal32kRc => pcr
+                    .ctrl_32k_conf
+                    .modify(|_, w| unsafe { w.clk_32k_sel().bits(0) }),
+                RtcCalSel::RtcCal32kXtal => pcr
+                    .ctrl_32k_conf
+                    .modify(|_, w| unsafe { w.clk_32k_sel().bits(1) }),
+                RtcCalSel::RtcCal32kOscSlow => pcr
+                    .ctrl_32k_conf
+                    .modify(|_, w| unsafe { w.clk_32k_sel().bits(2) }),
+            }
+        }
+
+        // Enable requested clock (150k is always on)
+        // Some delay is required before the time is stable
+        // Only enable if originaly was disabled
+        // If clock is already on, do nothing
+
+        let dig_32k_xtal_enabled = lp_clkrst.clk_to_hp.read().icg_hp_xtal32k().bit_is_set();
+
+        if cal_clk == RtcCalSel::RtcCal32kXtal && !dig_32k_xtal_enabled {
+            lp_clkrst
+                .clk_to_hp
+                .modify(|_, w| w.icg_hp_xtal32k().set_bit());
+        }
+
+        // TODO: very hacky
```

Comment:
What is this comment referring to? Can we get an explanation as to why the following is hacky and what needs to be done to improve it in the future?